aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,283 @@
+
+.. list-table:: Inline Route Constraints
+	:header-rows: 1
+
+	* - constraint
+	  - Example
+	  - Example Match
+	  - Notes
+	* - ``int``
+	  - {id:int}
+	  - 123
+	  - Matches any integer
+	* - ``bool``
+	  - {active:bool}
+	  - true
+	  - Matches ``true`` or ``false``
+	* - ``datetime``
+	  - {dob:datetime}
+	  - 2016-01-01
+	  - Matches a valid ``DateTime`` value (in the invariant culture - see `options <http://msdn.microsoft.com/en-us/library/aszyst2c(v=vs.110).aspx>`_)
+	* - ``decimal``
+	  - {price:decimal}
+	  - 49.99
+	  - Matches a valid ``decimal`` value
+	* - ``decimal``
+	  - {price:double}
+	  - 4.234
+	  - Matches a valid ``double`` value
+	* - ``float``
+	  - {price:float}
+	  - 3.14
+	  - Matches a valid ``float`` value
+	* - ``guid``
+	  - {id:guid}
+	  - 7342570B-44E7-471C-A267-947DD2A35BF9
+	  - Matches a valid ``Guid`` value
+	* - ``long``
+	  - {ticks:long}
+	  - 123456789
+	  - Matches a valid ``long`` value
+	* - ``minlength(value)``
+	  - {username:minlength(5)}
+	  - steve
+	  - String must be at least 5 characters long.
+	* - ``maxlength(value)``
+	  - {filename:maxlength(8)}
+	  - somefile
+	  - String must be no more than 8 characters long.
+	* - ``length(min,max)``
+	  - {filename:length(4,16)}
+	  - Somefile.txt
+	  - String must be at least 8 and no more than 16 characters long.
+	* - ``min(value)``
+	  - {age:min(18)}
+	  - 19
+	  - Value must be at least 18.
+	* - ``max(value)``
+	  - {age:max(120)}
+	  - 91
+	  - Value must be no more than 120.
+	* - ``range(min,max)``
+	  - {age:max(18,120)}
+	  - 91
+	  - Value must be at least 18 but no more than 120.
+	* - ``alpha``
+	  - {name:alpha}
+	  - Steve
+	  - String must consist of alphabetical characters.
+	* - ``regex(expression)``
+	  - {ssn:regex(\d{3}-\d{2}-\d{4})}
+	  - 123-45-6789
+	  - String must match the provided regular expression.
+	* - ``required``
+	  - {name:required}
+	  - Steve
+	  - Used to enforce that a non-parameter value is present during during URL generation.
+
+Inline constraints must match one of the above options, are an exception will be thrown.
+
+.. tip:: To constrain a parameter to a known set of possible values, you can use a regex: ``{action:regex(list|get|create)}``. This would only match the ``action`` route value to ``list``, ``get``, or ``create``. If passed into the constraints dictionary, the string "list|get|create" would be equivalent. Constraints that are passed in the constraints dictionary (not inline within a template) that don't match one of the known constraints are also treated as regular expressions.
+
+.. warning:: Using constraints for **validation** means that invalid input will result in a 404 (Not Found) instead of a 400 with an appropriate error message. Route constraints should be used to **disambiguate** between routes, not validate the inputs for a particular route.
```

Comment:
I would prefix this by saying "Avoid using constraints...." instead of "Using constraints...". This really is guidance